src/Engine/SpriteAnimator/SpriteAnimator.cpp

```diff
@@ -0,0 +1,70 @@
+#include "SpriteAnimator.h"
+
+SpriteAnimator::SpriteAnimator()
+    : initialized_(false)
+{}
+
+// Keeping this as an init right now because the objects may not
+int SpriteAnimator::init(SDL_Texture* texture, int rows, int cols, int on_every) {
+    //If there was no draw_order given as an argument, we will create a default
+    if (draw_order_.size() == 0) {
```

Comment:
Instead of having an if statement here, I think having the logic in the one with the draw_order, and this function will call the draw order func but pass in a draw order.

Reply:
sorry, in what "one with the draw_order"? can you rephrase this sentence? I don't understand